src/com/griddynamics/devops/mpl/Helper.groovy

```diff
@@ -126,19 +126,51 @@ abstract class Helper {
   }
 
   /**
-   * Cutting a stacktrace to just first execution of the module and one before
+   * Cut & simplify a stacktrace
    *
    * @param exception  container of the stacktrace
+   *
    * @return  List with stack trace elements
    */
   static StackTraceElement[] getModuleStack(Throwable exception) {
     List stack = exception.getStackTrace()
+
+    // For jenkins to remove starting trace items
+    if( stack.last()?.getFileName() == 'Thread.java' ) {
+      // Finding the first MPLModule call and cutting the trace
+      for( def i = stack.size(); i--; i > 0 ) {
```

Comment:
It should be: for( def i = l.size(); i>0; i-- ) right?

Reply:
Nope) Slightly different logic here.

Reply:
Ok, got it. Can we remove third statement? Otherwise it looks like an error.